bip-0183.md

```diff
@@ -0,0 +1,570 @@
+|----------------------------|------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| numleaves                  | varint                       | The number of leaves that was ever added to the accumulator at this block height. See [numleaves](bip-0181.md#Definitions)                                                                      |
+| target                     | varint                       | The position of the utreexo root in the optional accumulator of the utreexo roots                                                                                                                                |
+| blockhash                  | 32 byte vector               | The blockhash for this utreexo accumulator state                                                                                                                                                                 |
+| length of the root hashes  | varint                       | The length of the root hashes                                                                                                                                                                                    |
+| root hashes                | vector of 32 byte hashes     | The utreexo roots for the UTXO set at the blockhash. See [roots](bip-0181.md#Definitions)                                                                                                       |
+| length of the proof hashes | varint                       | The length of the proof hashes                                                                                                                                                                                   |
+| proof hashes               | vector of 32 byte hashes     | The proof hashes needed to validate with the pre-committed utreexo accumulator of the utreexo roots                                                                                                              |
+
+This message is for implementing an out-of-order block validation node[^2] or softchains[^3].
+
+Because the size of the state needed to validate blocks is so small with Utreexo, nodes can perform IBD in parallel and out of order.
+
+For example, a computer could divide the task of validating 800,000 blocks into 100 tasks of 8,000 blocks each: blocks 1 through 800, 800 through 1600, 1600 through 2400, and so on.
+
+In order start the 1600 through 2400 IBD task, however, the node should know what the state of the utxo set is at block 1600, so that it can validate and modify the accumulator.
```

Comment:
```suggestion
For example, a computer could divide the task of validating 800,000 blocks into 100 tasks of 8,000 blocks each: blocks 1 through 800, 801 through 1600, 1601 through 2400, and so on.

In order start the 1601 through 2400 IBD task, however, the node should know what the state of the UTXO set is at block 1600, so that it can validate and modify the accumulator.
```